src/lib/cost_model/cost_model_linear.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <map>
+#include <vector>
+
+#include "abstract_cost_model.hpp"
+#include "cost_feature.hpp"
+
+namespace opossum {
+
+/**
+ * CostModelLinear has different models for different kind of TableScans.
+ */
+enum class CostModelRuntimeTableScanType {
+  ColumnValueNumeric,
+  ColumnColumnNumeric,
+  ColumnValueString,
+  ColumnColumnString,
+  Like
+};
+
+/**
+ * Weights of the CostModelLinear for a particular build type (release, debug)
+ */
+struct CostModelRuntimeConfig final {
+  std::map<CostModelRuntimeTableScanType, CostFeatureWeights> table_scan_models;
+  std::map<OperatorType, CostFeatureWeights> other_operator_models;
+};
+
+/**
+ * Experimental Cost Model that tries to predict the actual runtime in microseconds of an operator. Experiments have
```

Comment:
I spoke to @mrks about this and we came to the conclusion that it would be best to include the foundation and interfaces for a cost model, but not the particular implementation since it is calibrated for a single machine, hard to reproduce and will not be adjusted in the near future. However, we see the need for a cost model in general for the sake of your master's thesis' progress. Do you think that's a viable approach?

Reply:
let's discuss that in person and keep the CostModelRuntime out of this review for the moment